x-pack/filebeat/input/httpjson/input_test.go

```diff
@@ -292,6 +293,49 @@ func TestInput(t *testing.T) {
 				`{"@timestamp":"2002-10-02T15:00:02Z","foo":"bar"}`,
 			},
 		},
+		{
+			name: "Test filename truncation",
+			setupServer: func(t *testing.T, h http.HandlerFunc, config map[string]interface{}) {
+				registerRequestTransforms()
+				t.Cleanup(func() { registeredTransforms = newRegistry() })
+				// mock timeNow func to return a fixed value
+				timeNow = func() time.Time {
+					t, _ := time.Parse(time.RFC3339, "2002-10-02T15:00:00Z")
+					return t
+				}
+
+				server := httptest.NewServer(h)
+				config["request.url"] = server.URL
+				t.Cleanup(server.Close)
+				t.Cleanup(func() { timeNow = time.Now })
+			},
+			baseConfig: map[string]interface{}{
+				"interval":       1,
+				"request.method": http.MethodGet,
+				"request.transforms": []interface{}{
+					map[string]interface{}{
+						"set": map[string]interface{}{
+							"target":  "url.params.$filter",
+							"value":   "alertCreationTime ge [[.cursor.timestamp]]",
+							"default": `alertCreationTime ge [[formatDate (now (parseDuration "-10m")) "2006-01-02T15:04:05Z"]]`,
+						},
+					},
+				},
+				"cursor": map[string]interface{}{
+					"timestamp": map[string]interface{}{
+						"value": `[[index .last_response.body "@timestamp"]]`,
+					},
+				},
+				"request.tracer.filename": "../../logs/httpjson/http-request-trace-*.ndjson",
```

Comment:
Is this file going into the repo? The repo should be left in a clean state after running the tests IMO. Please write this somewhere beneath `t.TempDir()` so it gets cleaned up automatically.